### PR TITLE
[Scheduling Sample] Example Data Bundle

### LINKED ISSUE #4941: Example data bundles
- Appointment Bundle
  - Appointment resource
  - Patient resource
  - Practitioner resource
- Patient History 
  - Patient resource
  - Multiple Appointment resources (past and future)
  - Multiple Encounter resources (linked to past appointments)

### LINKED ISSUE #4942: [Scheduling Sample] Project README

### PR DESCRIPTION
**Pull request by pamella**
## Description

- Resolves #4941
  - Implemented a bot to generate the example data bundle, addressing the need for dynamic content that cannot be achieved with a static JSON file.
    - The bot creates both past and upcoming slots based on the current date, ensuring that the data remains relevant and accurate over time.
- Resolves #4942

## Steps to test

- [ ] Build the bots: `npm run build:bots`
- [ ] Upload core data
- [ ] Upload example bots
- [ ] Upload example data
- [ ] Check if the resources are created properly
  - [ ] 2 Patients: Homer and Marge
  - [ ] 46 Slots: free, busy (attached to appointments), busy-unavailable (blocked)
  - [ ] 5 Appointments (1 canceled, 1 fulfilled, 3 booked)
  - [ ] 1 Encounter (for the fulfilled appointment)

[Screencast from 29-08-2024 14:30:36.webm](https://github.com/user-attachments/assets/f2bd6a73-c57f-4233-8bad-221807676e1e)

### CODE REVIEW

File: examples/medplum-scheduling-demo/package.json
Comment (pamella) [inline examples/medplum-scheduling-demo/package.json:1]: Automatically changed by `npm`.

File: examples/medplum-scheduling-demo/src/bots/example/example-data.ts [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+  createWeekSlots(today, -1, entries, scheduleReference, practitionerReference, [margeReference, homerReference]);
+  createWeekSlots(today, 0, entries, scheduleReference, practitionerReference, [margeReference, homerReference]);
+  createWeekSlots(today, 1, entries, scheduleReference, practitionerReference, [margeReference, homerReference]);
+
+  const bundle: Bundle = {
+    resourceType: 'Bundle',
+    type: 'batch',
+    entry: entries,
+  };
+  return bundle;
```
Comment (pamella) [inline examples/medplum-scheduling-demo/src/bots/example/example-data.ts:44]: I'm returning the bundle in the bot so that the `uploadExampleData` function aligns with the approach used in other demo apps. Does this seem reasonable, or would it be better to run `medplum.executeBatch` here instead?

File: examples/medplum-scheduling-demo/src/components/AppointmentDetails.tsx [hunk 2]
```diff
@@ -99,6 +99,8 @@ export function AppointmentDetails(props: AppointmentDetailsProps): JSX.Element
           <SearchControl
             search={encountersSearch}
             onChange={(e) => setEncountersSearch(e.definition)}
+            onClick={(e) => navigate(`/${e.resource.resourceType}/${e.resource.id}`)}
+            onAuxClick={(e) => window.open(`/${e.resource.resourceType}/${e.resource.id}`, '_blank')}
```
Comment (pamella) [inline examples/medplum-scheduling-demo/src/components/AppointmentDetails.tsx:103]: Drive-by fix: Now it's possible to redirect the uset to the encounter detail page.

File: examples/medplum-scheduling-demo/src/pages/UploadDataPage.tsx [hunk 3]
```diff
@@ -135,11 +138,34 @@ function checkBotsUploaded(medplum: MedplumClient): boolean {
...
     return true;
   }
   return false;
 }
+
+async function uploadExampleData(medplum: MedplumClient, profile: Practitioner): Promise<void> {
+  // Unlike the other bundles, which source data from JSON files, this example data bundle is
+  // provided by a bot to handle dynamic content.
+  const batch: Bundle = await medplum.executeBot({ system: 'http://example.com', value: 'example-data' }, profile);
+  const result = await medplum.executeBatch(batch);
```
Comment (pamella) [inline examples/medplum-scheduling-demo/src/pages/UploadDataPage.tsx:156]: That's the code comment that explains why we are not using a static JSON file for the example data of this demo app.

Comment (pamella) [inline examples/medplum-scheduling-demo/src/pages/UploadDataPage.tsx:156]: That's the code comment that explains why we are not using a static JSON file for the example data of this demo app.

**Review by rahul1 — APPROVED** [review]
